src/d3a/d3a_core/sim_results/export_unmatched_loads.py

```diff
@@ -194,3 +212,81 @@ def arrange_output(self, indict, area):
                         indict[area.name][child.name] = indict[child.name]
                     self.arrange_output(indict, child)
         return indict
+
+
+class MarketUnmatchedLoads:
+
+    def __init__(self):
+        self._unmatched_loads_incremental = {}
+        self._unmatched_loads_incremental_uuid = {}
+
+    def _merge_base_area_unmatched_loads(self, accumulated_results, current_results, area):
+        for target, target_value in current_results[area].items():
+            if target not in accumulated_results[area]:
+                accumulated_results[area][target] = deepcopy(target_value)
+            else:
+                if target == 'type':
+                    continue
+                elif target == 'unmatched_loads':
+                    self._merge_accumulated_unmatched_loads(
+                        accumulated_results, current_results, area
+                    )
+                else:
+                    self._merge_target_area_unmatched_loads(
+                        accumulated_results, current_results, area, target
+                    )
+
+    def _merge_target_area_unmatched_loads(self, accumulated_results, current_results,
+                                           area, target):
+        target_ul = accumulated_results[area][target]['unmatched_loads']
+        current_ul = current_results[area][target]['unmatched_loads']
+        for timestamp, ts_value in current_ul.items():
+            if timestamp not in target_ul:
+                target_ul[timestamp] = deepcopy(ts_value)
+            else:
+                if 'unmatched_times' not in current_ul[timestamp]:
+                    continue
+                if 'unmatched_times' not in target_ul[timestamp]:
+                    target_ul[timestamp]['unmatched_times'] = {}
+                for device, time_list in current_ul[timestamp]['unmatched_times'].items():
+                    if device not in target_ul[timestamp]['unmatched_times']:
+                        target_ul[timestamp]['unmatched_times'][device] = deepcopy(time_list)
+                    else:
+                        for ts in time_list:
+                            if ts not in target_ul[timestamp]['unmatched_times'][device]:
+                                target_ul[timestamp]['unmatched_times'][device].append(ts)
+
+                unm_count = 0
+                for _, hours in target_ul[timestamp]['unmatched_times'].items():
+                    unm_count += len(hours)
+
+                    target_ul['unmatched_count'] = unm_count
+
+    def _merge_accumulated_unmatched_loads(self, accumulated_results, current_results, area):
```

Comment:
Just minor, but doesnt this function only copies the value from the current to the accumulated_results? 
The name merge confused me a bit. Again, just personal taste. If you dont agree, just leave it.

Reply:
Agreedm changing it to copy_accumulated_unmatched_loads

Reply:
I tried to leave at least very short doc strings, about what the individual functions do in the ExportUnmatchedLoads class. In my experience these can help later when understanding the code, especially if it is for these recursive structures. I suggest to do it also for the MarketUnmatchedLoads. Thanks a lot!

Reply:
ok, in that case I will add some docstrings for this class too, thanks!